### PR TITLE
Add scan binaries template

### PR DESCRIPTION
**Pull request by mmsmits**

### CODE REVIEW

File: scanBinaries.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+parameters: 
+  artifactName:  # name of the artifact on Azure DevOps that contains the Zip file                         
+  zipfile: # name or pattern of the zipfile to be extracted and scanned
+  directory: # directory where the zipfile is extracted
```
Comment (marcovisserFurore) [inline scanBinaries.yml:4]: Is it really necessary that `directory` is a parameter? That could also be a variable in this template, right? Or is this directory used elsewhere outside this stage?

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
I think we can have 1 parameter less.

File: scanBinaries.yml (hunk 1 above)
Comment (mmsmits) [reply scanBinaries.yml:4]: Ah you are right, we can just call it "binaries" or something common

File: scanBinaries.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - task: DownloadPipelineArtifact@2
+      displayName: Download ZipDeployArchive
+      inputs:
+        artifactName: ${{ parameters.artifactName }}   
+        targetPath: '$(System.DefaultWorkingDirectory)'
+    - task: ExtractFiles@1
+      displayName: Extract  ${{ parameters.zipFile }}      
+      inputs:
+        archiveFilePatterns: ${{ parameters.zipFile }}   
+        destinationFolder: ${{ parameters.directory }}
```
Comment (marcovisserFurore) [inline scanBinaries.yml:29]: This is not going to work. You have to change this to the new variable.

File: scanBinaries.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        targetPath: '$(System.DefaultWorkingDirectory)'
+    - task: ExtractFiles@1
+      displayName: Extract  ${{ parameters.zipFile }}      
+      inputs:
+        archiveFilePatterns: ${{ parameters.zipFile }}   
+        destinationFolder: ${{ parameters.directory }}   
+        cleanDestinationFolder: true
+        overwriteExistingFiles: false
+    - script: | 
+        docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v /tmp:/tmp -v ${{ parameters.directory }}:/src  aquasec/trivy:latest  --exit-code 1 --format table --scanners vuln,config,secret filesystem /src
```
Comment (marcovisserFurore) [inline scanBinaries.yml:33]: Same here

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
Some errors now because of the new variable.

File: scanBinaries.yml (hunk 2 above)
Comment (marcovisserFurore) [reply scanBinaries.yml:29]: So `$(directory)`. See also [here](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-variables-in-pipeline).

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]

**Review by marcovisserFurore — APPROVED** [review]

**Pull request merged**